dem/core/registry.py

```diff
@@ -3,32 +3,27 @@
 """
 # dem/core/registry.py
 
-import dxf
-import requests
 import json
-
-def auth(dxf_base_obj: dxf.DXFBase, response: requests.Response):
-    with open("../password", "r") as password:
-        password = password.read()
-        dxf_base_obj.authenticate("axemsolutions", password=password[:-1], response=response)
+import docker
+import subprocess
 
 def list_repos():
-    r = requests.get("https://hub.docker.com/v2/repositories/axemsolutions")
-    r_deserialized = json.loads(r.text)
-    repos = r_deserialized["results"]
+    registryimagelist = []
     images = []
-    for repo in repos:
-        for description in repo["description"].split(';'):
-            description = description.split(':')
-            print("\ttool name: " + description[0] + "\ttool version: " + description[1])
-        print("")
-
-        repo_path = "axemsolutions/" + repo["name"]
-        dxf_obj = dxf.DXF("registry-1.docker.io", repo_path, auth=auth)
 
-        for tag in dxf_obj.list_aliases():
-            images.append(repo_path + ':' + tag)
+    docker_client = docker.from_env()
 
+    for repositories in docker_client.images.search("axemsolutions"):                
+        registryimagelist.append(repositories['name'])
```

Comment:
Please move this into the ContainerEngine class. The ContainerEngine should hide the actual container engine which is the Docker in this case.
The ContainerEngine should have a search method that returns with the repositiories.